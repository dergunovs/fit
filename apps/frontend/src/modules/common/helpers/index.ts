import { toast } from 'mhz-ui';

export async function copyToClipboard(text: string, toastText: string) {
  await navigator.clipboard.writeText(text);

  toast.success(toastText);
}
