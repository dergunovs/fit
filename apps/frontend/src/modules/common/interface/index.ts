import { TPostAuthLoginDTO } from 'fitness-tracker-contracts';
import { FunctionalComponent } from 'vue';

export interface INavItem {
  url: string;
  title: string;
  icon: FunctionalComponent;
}

export interface IOnSuccess {
  create?: (id?: string) => Promise<void>;
  update?: () => Promise<void>;
  delete?: () => Promise<void>;
  login?: (user: TPostAuthLoginDTO) => Promise<void>;
  setup?: () => Promise<void>;
  register?: () => Promise<void>;
}

export interface IPromptEvent extends Event {
  prompt: () => Promise<void>;
}

export interface IGetLatestVersionDTO {
  'dist-tags': { latest: string };
}
