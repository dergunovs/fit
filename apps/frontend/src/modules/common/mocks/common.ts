import { vi } from 'vitest';

export const spyCopyToClipboard = vi.spyOn(navigator.clipboard, 'writeText').mockResolvedValue();
