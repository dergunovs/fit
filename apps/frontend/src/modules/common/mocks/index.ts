export * from './helpers';
export * from './ui';
export * from './router';
export * from './layout';
export * from './pwa';
export * from './service';
