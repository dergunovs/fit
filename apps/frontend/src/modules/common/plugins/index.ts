import { createI18n, I18nOptions } from 'vue-i18n';

import ru from '@/common/locales/ru.json';
import en from '@/common/locales/en.json';

const messages = { en, ru } as const;

const options: I18nOptions = { legacy: false, locale: 'ru', messages };

// eslint-disable-next-line @typescript-eslint/no-type-alias
export type TMessageSchema = typeof en;

export const i18n = createI18n<false, typeof options>(options);
