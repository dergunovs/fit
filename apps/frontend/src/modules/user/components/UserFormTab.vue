<template>
  <UiFlex column gap="16">
    <h3 data-test="user-form-tab-title">{{ props.title }}</h3>

    <p data-test="user-form-tab-description">{{ props.description }}</p>

    <slot></slot>
  </UiFlex>
</template>

<script setup lang="ts">
import { UiFlex } from 'mhz-ui';

interface IProps {
  title: string;
  description: string;
}

const props = defineProps<IProps>();
</script>
