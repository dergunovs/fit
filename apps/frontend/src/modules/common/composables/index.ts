import { ref, computed, onBeforeMount, Ref } from 'vue';
import { useRouter, useRoute } from 'vue-router';
import { useI18n } from 'vue-i18n';
import { TLocale } from 'fitness-tracker-contracts';

import LayoutAdmin from '@/common/components/LayoutAdmin.vue';
import LayoutDefault from '@/common/components/LayoutDefault.vue';

import IconExercise from '@/common/icons/exercise.svg?component';
import IconActivity from '@/common/icons/activity.svg?component';
import IconEquipment from '@/common/icons/weight.svg?component';
import IconMuscle from '@/common/icons/muscle.svg?component';
import IconUser from '@/common/icons/user.svg?component';

import { URL_USER, URL_USER_PROFILE } from '@/user/constants';
import { URL_EXERCISE } from '@/exercise/constants';
import { URL_ACTIVITY_ADMIN, URL_ACTIVITY_CREATE } from '@/activity/constants';
import { URL_EQUIPMENT } from '@/equipment/constants';
import { URL_MUSCLE } from '@/muscle/constants';
import { URL_HOME } from '@/common/constants';
import { INavItem } from '@/common/interface';

export function useLayout() {
  const router = useRouter();
  const route = useRoute();

  const isLoaded = ref(false);

  const layoutComponent = computed(() => (route.meta.layout === 'admin' ? LayoutAdmin : LayoutDefault));

  onBeforeMount(async () => {
    await router.isReady();
    isLoaded.value = true;
  });

  return {
    isLoaded,
    layoutComponent,
  };
}

export function useTI18n() {
  const { t, tm, rt, availableLocales, locale } = useI18n();

  return { t, tm, rt, availableLocales: availableLocales as TLocale[], locale: locale as Ref<TLocale> };
}

export function useLocale() {
  const { locale, availableLocales } = useTI18n();

  function toggleLocale() {
    locale.value = availableLocales.find((l) => l !== locale.value) || locale.value;

    localStorage.setItem('locale', locale.value);
  }

  function initLocale() {
    if (localStorage.getItem('locale')) {
      locale.value = localStorage.locale;
    } else {
      const defaultLanguage = Intl.DateTimeFormat().resolvedOptions().locale.slice(0, 2) as TLocale;

      const isLanguageSupported = availableLocales.includes(defaultLanguage);

      if (isLanguageSupported) locale.value = defaultLanguage;
    }
  }

  onBeforeMount(() => initLocale());

  return {
    toggleLocale,
    initLocale,
  };
}

export function useNavItems() {
  const { t } = useTI18n();

  const NAV_ITEMS = computed<INavItem[]>(() => [
    { url: URL_EXERCISE, title: t('exercise.many'), icon: IconExercise },
    { url: URL_ACTIVITY_ADMIN, title: t('activity.many'), icon: IconActivity },
    { url: URL_EQUIPMENT, title: t('equipment.one'), icon: IconEquipment },
    { url: URL_MUSCLE, title: t('muscle.many'), icon: IconMuscle },
    { url: URL_USER, title: t('user.many'), icon: IconUser },
  ]);

  const BOTTOM_NAV_ITEMS = computed<INavItem[]>(() => [
    { url: URL_HOME, title: t('statistics'), icon: IconActivity },
    { url: URL_ACTIVITY_CREATE, title: t('activity.one'), icon: IconExercise },
    { url: URL_USER_PROFILE, title: t('profile'), icon: IconUser },
  ]);

  return { NAV_ITEMS, BOTTOM_NAV_ITEMS };
}
