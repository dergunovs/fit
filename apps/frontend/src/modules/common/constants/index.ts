import IconExercise from '@/common/icons/nav-exercise.svg?component';
import IconActivity from '@/common/icons/nav-activity.svg?component';
import IconEquipment from '@/common/icons/weight.svg?component';
import IconUser from '@/common/icons/nav-user.svg?component';

import { URL_USER } from '@/user/constants';
import { URL_EXERCISE } from '@/exercise/constants';
import { URL_ACTIVITY_ADMIN } from '@/activity/constants';
import { URL_EQUIPMENT } from '@/equipment/constants';
import { INavItem } from '@/common/interface';
import { URL_SETUP } from '@/auth/constants';

export const NAV_ITEMS: INavItem[] = [
  { _id: '1', url: URL_EXERCISE, title: 'Упражнения', icon: IconExercise },
  { _id: '2', url: URL_ACTIVITY_ADMIN, title: 'Занятия', icon: IconActivity },
  { _id: '3', url: URL_EQUIPMENT, title: 'Оборудование', icon: IconEquipment },
  { _id: '4', url: URL_USER, title: 'Пользователи', icon: IconUser },
];

export const UPDATE_BUTTON_TEXT = 'Обновить';
export const CREATE_BUTTON_TEXT = 'Создать';

export const URL_HOME = '/';
export const URL_ERROR = '/404';

export const GUEST_PAGES = [URL_HOME, URL_SETUP];
