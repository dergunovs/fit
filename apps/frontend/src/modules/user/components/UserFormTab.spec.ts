import { describe, it, expect, beforeEach, afterEach } from 'vitest';
import { VueWrapper, enableAutoUnmount } from '@vue/test-utils';
import { dataTest } from 'mhz-helpers';

import UserFormTab from './UserFormTab.vue';

import { wrapperFactory } from '@/common/test';

const tabTitle = dataTest('user-form-tab-title');

let wrapper: VueWrapper<InstanceType<typeof UserFormTab>>;

const title = 'заголовок';
const description = 'описание';

beforeEach(() => {
  wrapper = wrapperFactory(UserFormTab, { title, description });
});

enableAutoUnmount(afterEach);

describe('UserFormTab', async () => {
  it('exists', async () => {
    expect(wrapper.findComponent(UserFormTab)).toBeTruthy();
  });

  it('matches snapshot', async () => {
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('shows title', async () => {
    expect(wrapper.find(tabTitle).text()).toStrictEqual(title);
  });
});
