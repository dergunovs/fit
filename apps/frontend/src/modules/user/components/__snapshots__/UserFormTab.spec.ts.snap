// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`UserFormTab > matches snapshot 1`] = `
"<div column="" gap="16">
  <h3 data-test="user-form-tab-title">заголовок</h3>
  <p data-test="user-form-tab-description">описание</p>
</div>"
`;
